lib/utils/either.dart

```diff
@@ -0,0 +1,21 @@
+abstract class Either<L, R> {
+  const Either();
+
+  bool get success => this is Right<L, R>;
+  bool get failure => this is Left<L, R>;
```

Comment:
To follow the usual implementation pattern of Either, I suggest these should be called `isLeft` and `isRight`